nixos/modules/services/web-servers/traefik.nix

```diff
@@ -4,20 +4,18 @@ with lib;
 
 let
   cfg = config.services.traefik;
-  jsonValue = with types;
-    let
-      valueType = nullOr (oneOf [
-        bool
-        int
-        float
-        str
-        (lazyAttrsOf valueType)
-        (listOf valueType)
-      ]) // {
-        description = "JSON value";
-        emptyValue.value = { };
-      };
-    in valueType;
+  jsonValue = with types; recursive (t:
+    nullOr (oneOf [
+      bool
+      int
+      float
+      str
+      (lazyAttrsOf t)
```

Comment:
Off-topic: choice between `attrsOf` and `lazyAttrsOf` should probably be a `formats.json` parameter.